sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -76,6 +76,24 @@ abstract class DataType {
     DataType.equalsIgnoreNullability(this, other)
 
   /**
+   * Check if `this` is a primitive data type.
+   */
+  def isPrimitive: Boolean = {
```

Comment:
I didn't quite get your point.
But I think I can simplify the code by using 'isInstanceOf[ArrayType]' ?
I added that method because, it was present in the documentation and is not implemented in 'DataType.scala'.